### PR TITLE
Worker Init/Final Log on Datapoint

### PR DESCRIPTION
**Pull request by nllong**
* Add set_gem_version.rb to the repo (for versioning and ease). This version can work locally in development or in production but must use resque.
* Display the contents of the worker init/final script via API/HTML
* Add simple test for running init scripts (needs more work)

Upon merging have [this page](https://github.com/NREL/OpenStudio-server/wiki/Gem-Bundle-used-by-OpenStudio) reference the script in the PR.

### CODE REVIEW

File: doc/initialization_scripts/set_gem_version.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+if [ ! -f "$GEMFILE_DIR/openstudio-gems.gemspec" ]; then
+  echo "openstudio-gems.gemspec does not exist in: ${GEMFILE_DIR}"
+  echo "!!! This script only works with OpenStudio 2.8.0 and newer !!!"
+  exit 1
+fi
+
+# First check if there is a file that indicates the gem has already been updated.
+# We only need to update the bundle once / worker, not every time a data point is initialized.
+echo "Checking if Gemfile has been updated in ${GEMFILEUPDATE}"
```
Comment (anyaelena) [inline doc/initialization_scripts/set_gem_version.sh]: should we do this at the start?

File: server/app/jobs/dj_jobs/run_simulate_data_point.rb [hunk 2]
```diff
@@ -540,6 +540,11 @@ def run_script_with_args(script_name)
       msg = "Error #{e.message} running #{script_name}: #{e.backtrace.join("\n")}"
       @sim_logger.error msg
       raise msg
+    ensure
+      # save the log information to the datapoint if it exists
+      if File.exist? log_path
+        @data_point.worker_logs[script_name] = File.read(log_path).lines
```
Comment (anyaelena) [inline server/app/jobs/dj_jobs/run_simulate_data_point.rb]: i like this update.  the logs are also downloaded w/data point zip file.

File: server/spec/features/resque_run_batch_datapoints_spec.rb [hunk 3]
```diff
@@ -49,8 +49,12 @@
 
   before do
     # Look at DatabaseCleaner gem in the future to deal with this.
-    Project.destroy_all
-    Delayed::Job.destroy_all
+    begin
+      Project.destroy_all
+      Delayed::Job.destroy_all
+    rescue Errno::EACCES => e
+      puts "Cannot unlink files, will try and continue"
```
Comment (anyaelena) [inline server/spec/features/resque_run_batch_datapoints_spec.rb]: nice!

**Comment by anyaelena** [comment]
Thanks @nllong!

File: doc/initialization_scripts/set_gem_version.sh (hunk 1 above)
Comment (nllong) [reply doc/initialization_scripts/set_gem_version.sh]: yeah, that would make more sense.

**Pull request merged**